### PR TITLE
Minimize boilerplate code around Rocks column families

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
Reducing the clutter / repeated use of constants for column families. Also added some instructions for what to update when a new column is added.

### CODE REVIEW

File: ledger/src/blockstore_db.rs [hunk 1]
```diff
@@ -288,121 +294,42 @@ impl Rocks {
...
+            new_cf_descriptor::<ShredCode>(&access_type, &oldest_slot),
+            new_cf_descriptor::<TransactionStatus>(&access_type, &oldest_slot),
+            new_cf_descriptor::<AddressSignatures>(&access_type, &oldest_slot),
+            new_cf_descriptor::<TransactionMemos>(&access_type, &oldest_slot),
+            new_cf_descriptor::<TransactionStatusIndex>(&access_type, &oldest_slot),
+            new_cf_descriptor::<Rewards>(&access_type, &oldest_slot),
+            new_cf_descriptor::<Blocktime>(&access_type, &oldest_slot),
+            new_cf_descriptor::<PerfSamples>(&access_type, &oldest_slot),
+            new_cf_descriptor::<BlockHeight>(&access_type, &oldest_slot),
+            new_cf_descriptor::<ProgramCosts>(&access_type, &oldest_slot),
```
Comment (yhchiang-sol) [inline ledger/src/blockstore_db.rs:317]: This is much cleaner!  Thanks for the re-work!

File: ledger/src/blockstore_db.rs [hunk 2]
```diff
@@ -117,6 +117,12 @@ pub enum IteratorMode<Index> {
     From(Index, IteratorDirection),
 }
 
+// When adding a new column ...
+// - Add struct below and implement `Column` and `ColumnName` traits
+// - Add descriptor in Rocks::open() and name in Rocks::columns()
+// - Account for column in both `run_purge_with_stats()` and
+//   `compact_storage()` in ledger/src/blockstore/blockstore_purge.rs !!
+// - Account for column in `analyze_storage()` in ledger-tool/src/main.rs
```
Comment (yhchiang-sol) [inline ledger/src/blockstore_db.rs:125]: Use `///` for function comment block as we previously discussed :p.

**Review by yhchiang-sol — APPROVED** [review]
LGTM!  Only nit is to use `///` for function comment block.

File: ledger/src/blockstore_db.rs (hunk 2 above)
Comment (steviez) [reply ledger/src/blockstore_db.rs]: Yeah, agreed that it is more internal guidelines than public facing docs that somebody would be interested; I'll throw it inside the the `columns` mod tho!

**Pull request merged**